packages/flutter_tools/lib/src/drive/drive_service.dart

```diff
@@ -266,10 +266,15 @@ class FlutterDriverService extends DriverService {
       // When profiling memory the original launch future will never complete.
       await _devtoolsLauncher.processStart;
     }
+    if (arguments?.isNotEmpty == true && packageConfig['test'] == null) {
+      _logger.printError('Arguments $arguments will be ignored because there is no dependency on package:test in the pubspec.yaml.');
+    }
     try {
       final int result = await _processUtils.stream(<String>[
         _dartSdkPath,
```

Comment:
we can't use `dart test` because it breaks the contract of `drive` in that it always requires an up to date pubspec/packages file. This broke a number of projects in the plugins repo

Reply:
We're conditionally using it - only if the pubspec has a depdency on test.

See line 275

Reply:
That is the change I tried to make before and it broke several tests on plugins CI

Reply:
also if you specify `--no-pub` or run from a directory other than your project this change will break it

Reply:
So you might need a `--super-secret-use-test-exe` argument as an opt-in

Reply:
Argh. Ok.

In that case, we need to specify timeouts everywhere you write a `test`, otherwise it's 30 seconds.

Reply:
I'll revert these changes and apply those for now.